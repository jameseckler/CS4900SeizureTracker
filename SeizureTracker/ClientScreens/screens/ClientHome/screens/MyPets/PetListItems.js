import React, { Component } from 'react';
import { Dimensions, StyleSheet, TouchableOpacity, Text, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';
import {w, h, totalSize} from "../../../../../FirebaseLogin/api/Dimensions";
import Icon from 'react-native-vector-icons/Ionicons';

/*
  Acts as a touchable button for displaying each pet in the PetList page.
  Displays a pet's name and passes properties through it to the function called
  onPress. Will link to Pet Info page with all pet properties passed to that page 
  for display.
*/
export default class CreateLogButton extends Component {

  // Renders Touchable Opacity button 
  render() {
    return (
      <TouchableOpacity
        activeOpacity={0.5}
        onPress={this.props.click}
        style={styles.button}>
        <Icon style={{marginLeft: w(3.5)}} name="md-paw" color = 'white' size={23} />
        <Text style={styles.text}>{this.props.name}</Text>
      </TouchableOpacity>
    );
  }
}

CreateLogButton.propTypes = {
  click: PropTypes.func.isRequired,
  age: PropTypes.string.isRequired,
  breed: PropTypes.string.isRequired,
  dateString: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  sex: PropTypes.string.isRequired,
  symptoms: PropTypes.string.isRequired,
  weight: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
  button: {
    width: w(80),
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#101d26',
    paddingVertical: w(2),
    borderRadius: w(0),
    borderColor: '#E0E0E0',
    borderWidth: 1,
    marginVertical: h(1),
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'flex-start',
    marginLeft: w(3) 
  },
  spinner: {
    height: h(5),
  },
  text: {
    color: 'white',
    fontWeight: '500',
    paddingVertical: h(1),
    fontSize: totalSize(3.5),
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'flex-start',
    marginLeft: w(5)
  }
});