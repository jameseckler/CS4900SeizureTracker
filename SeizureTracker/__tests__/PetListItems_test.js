import React from 'react';
import renderer from 'react-test-renderer';
import PetListItems from '../ClientScreens/screens/ClientHome/screens/MyPets/PetListItems';

test ("Pet List Items render properly", () => {
    
    const rendered = renderer.create(<PetListItems click={()=> this.props.navigation.navigate('ClientHome')} text='Test pet'/>).toJSON();
    expect(rendered).toBeTruthy();

});

it ("Test Pet List Items against snapshot", () => {

    const tree = renderer.create(<PetListItems click={()=> this.props.navigation.navigate('ClientHome')} text='Test pet'/>).toJSON();
    expect(tree).toMatchSnapshot();

});