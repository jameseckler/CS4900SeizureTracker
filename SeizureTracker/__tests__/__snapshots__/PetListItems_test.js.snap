// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pet List Items against snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#101d26",
      "borderColor": "#E0E0E0",
      "borderRadius": 0,
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginLeft": 37.5,
      "marginVertical": 13.34,
      "opacity": 1,
      "paddingVertical": 15,
      "width": 675,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "white",
        "flex": 1,
        "flexDirection": "row",
        "fontSize": 53.56322059025204,
        "fontWeight": "500",
        "justifyContent": "flex-start",
        "marginLeft": 37.5,
        "paddingVertical": 13.34,
      }
    }
  >
    Test pet
  </Text>
</View>
`;
